### PR TITLE
Feat/75 Add project icon upload

### PR DESCRIPTION
**Pull request by reglim**
I created an API method to upload a project icon and documented it in getting-started.md.

I haven't created a UI yet, as this doesn't make sense considering the planned migration to vue 3.

fixes: #75

### CODE REVIEW

**Comment by fliiiix** [comment]
should be one commit otherwise lgtm

File: doc/getting-started.md [hunk 1]
```diff
@@ -84,4 +84,15 @@ If you want to remove the version `1.0.0` from awesome-project.
 
 ```sh
 curl -X DELETE --header "Docat-Api-Key: <token>" http://localhost:8000/api/awesome-project/1.0.0
+```
+
+#### Upload Project Icon
+
+To upload a icon (only jpegs are supported), you don't need a token, except if
```
Comment (randombenj) [inline doc/getting-started.md:91]: maybe add an issue that we should support more formats in the future?

File: docat/tests/test_upload_icon.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+
+def test_successful_icon_upload(client_with_claimed_project):
+    upload_folder_response = client_with_claimed_project.post(
+        "/api/some-project/1.0.0", files={"file": ("index.html", io.BytesIO(b"<h1>Hello World</h1>"), "plain/text")}
+    )
+    assert upload_folder_response.status_code == 201
+
+    with patch("shutil.copyfileobj") as copyfileobj_mock:
+        with patch("os.remove") as remove_file_mock:
```
Comment (randombenj) [inline docat/tests/test_upload_icon.py:14]: Try not to nest `with patch` but rather use `with patch("") as patch1, patch("") as patch2`

File: doc/getting-started.md (hunk 1 above)
Comment (fliiiix) [reply doc/getting-started.md:91]: idk lets keep it simple and just allow one file format should be enough? but then maybe we should go for png as the only supported file format?

Comment (randombenj) [reply doc/getting-started.md:91]: Exactly, I don't see a good reason why png for example is not supported

File: docat/tests/test_upload_icon.py (hunk 2 above)
Comment (reglim) [reply docat/tests/test_upload_icon.py:14]: I searched for something like this but didn't find it at first :+1:

File: doc/getting-started.md (hunk 1 above)
Comment (reglim) [reply doc/getting-started.md:91]: I just used what was there, the project repository looks like this so I fitted it to what was there:

```
  /**
   * Returns the logo URL of a given project
   * @param {string} projectName Name of the project
   */
  getProjectLogoURL(projectName) {
    return `${this.baseURL}/${resource}/${projectName}/logo.jpg`
  },

```

Comment (reglim) [reply doc/getting-started.md]: Ok, I changed it to PNGs

**Pull request closed without merging**